src/pages/t/admin/teachers.tsx

```diff
@@ -87,127 +102,88 @@ const Teachers: NextPage<{ allTeachers: Array<Teacher> }> = ({
         show={showEdit}
         onClose={() => setShowEdit(false)}
         // TODO: Refetch teachers here ↓
-        onSubmit={() => setShowEdit(false)}
+        onSubmit={() => {
+          setShowEdit(false);
+          router.replace(router.asPath);
+        }}
         mode="edit"
         person={editingPerson}
       />
       <EditPersonDialog
         show={showAdd}
         onClose={() => setShowAdd(false)}
         // TODO: Refetch teachers here ↓
```

Comment:
You can remove TODOs when you're done with them, I honestly think refreshing the page is…good enough.